NeonGeckoCom/neon_debos
# feat: Mark 2 Audio Receiver

Closes #16 

New support:
- [raspotify](https://dtcooper.github.io/raspotify/) for Spotify Connect
- [UxPlay](https://github.com/FDH2/UxPlay) for AirPlay support
- [KDE Connect](https://kdeconnect.kde.org/) daemon, which was available but hidden in a funny place. It will now run as a systemd service. There are also some configuration changes that expose OCP to KDE Connect. Unfortunately we can't cast audio to it, but audio can be controlled by KDE Connect now. Note that this is only available via Linux, Android, and Windows KDE Connect apps - iOS does not support Media Players.
- Bluetooth Speaker capabilities

For pairing of KDE Connect and Bluetooth, we use a combination admin PHAL plugin/skill combo:
https://github.com/mikejgray/skill-mark2-audio-receiver
https://github.com/mikejgray/neon-phal-plugin-audio-receiver

To test:

- Raspotify: Load Spotify, select Neon Mark 2, validate sound works as expected.
- UxPlay: From an Apple device, select uxplay@neon as your AirPlay device and validate sound works as expected. I understand it's possible to do something like this from Android but I wasn't able to work out how with my meager Android skills and slow tablet.
- KDE Connect: Open KDE Connect on your device of choice. From the Mark 2, type: `/usr/local/bin/autopair-kdeconnect.sh`. Pair as usual from your device. Test various features, like file sharing and media player control.
- Bluetooth: Validate that the `bluetooth` service is running. To pair, run `/usr/local/bin/autopair-bluetooth.sh`. The Mark 2 should function as a Bluetooth speaker.

The PHAL plugin/skill combination will allow you to do all of the above by voice, as well as enable/disable each service, rename devices in Raspotify and KDE Connect, and handle autopairing.
